oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexStatistics.java

```diff
@@ -219,9 +221,14 @@ static class CountCacheLoader extends CacheLoader<StatsRequestDescriptor, Intege
 
         @Override
         public @NotNull ListenableFuture<Integer> reload(@NotNull StatsRequestDescriptor crd, @NotNull Integer oldValue) {
-            ListenableFutureTask<Integer> task = ListenableFutureTask.create(() -> count(crd));
-            REFRESH_EXECUTOR.execute(task);
-            return task;
+            CompletableFuture<Integer> task = CompletableFuture.supplyAsync(() -> {
+                try {
+                    return count(crd);
+                } catch (IOException e) {
+                    throw new CompletionException(e);
+                }
+            }, REFRESH_EXECUTOR);
+            return FutureConverter.toListenableFuture(task);
```

Comment:
Need to return as `ListenableFuture` since we are overriding guava's `CacheLoader`'s method